frontend/src/v5/ui/routes/dashboard/projects/tickets/ticketsList/slides/ticketSlide.component.tsx

```diff
@@ -39,16 +39,27 @@ export const TicketSlide = ({ template, ticketId }: TicketSlideProps) => {
 	const { teamspace, project, containerOrFederation } = useParams<DashboardTicketsParams>();
 	const isFederation = modelIsFederation(containerOrFederation);
 	const ticket = TicketsHooksSelectors.selectTicketByIdRaw(containerOrFederation, ticketId);
+	const templateValidationSchema = getValidators(template);
 
 	const formData = useForm({
-		resolver: yupResolver(getValidators(template)),
+		resolver: yupResolver(templateValidationSchema),
 		mode: 'onChange',
 		defaultValues: ticket,
 	});
 
-	const onBlurHandler = () => {
-		const values = dirtyValues(formData.getValues(), formData.formState.dirtyFields);
-		const validVals = removeEmptyObjects(nullifyEmptyObjects(filterErrors(values, formData.formState.errors)));
+	const onBlurHandler = async () => {
+		const formValues = formData.getValues();
+		let errors = {};
+		try {
+			// cannot use formState.errors because the validation for complex objects is completed after
+			// onBlur gets called, so formState.errors for those objects is updated to the previous
+			// onBlur call instead and might claim there are no errors when it's not the case
+			await templateValidationSchema.validateSync(formValues, { abortEarly: false });
+		} catch (yupError) {
+			(yupError?.inner || []).forEach(({ path, message }) => set(errors, path, { message }));
+		}
+		const values = dirtyValues(formValues, formData.formState.dirtyFields);
+		const validVals = removeEmptyObjects(nullifyEmptyObjects(filterErrors(values, errors)));
```

Comment:
Is the `removeEmptyObjects` here doing anything? Surely there aren't any empty values after `nullifyEmptyObjects` is called.

As a side note (I know it's not part of this issue or anything you've done), but I don't think those two functions have accurate names if I'm understanding them correctly.

Reply:
> Is the `removeEmptyObjects` here doing anything? Surely there aren't any empty values after `nullifyEmptyObjects` is called.

I feel you are right, I checked and tested the app a bit to find out a possible scenario where `removeEmptyObjects` is being useful, but I couldn't find anything. Let's keep this as is for now and let's ask Santiago when he gets back as he's the one who added it

> As a side note (I know it's not part of this issue or anything you've done), but I don't think those two functions have accurate names if I'm understanding them correctly.

Why not? `nullifyEmptyObjects` claims to convert `{}` to `null`, whereas `removeEmptyObjects` claims to remove them, and so they do
![image](https://github.com/3drepo/3drepo.io/assets/43547703/58195bdc-0530-46ce-ac13-c12b74180412)

Reply:
Ah, if that's the case then the description provided above `nullifyEmptyObjects` is wrong
![image](https://github.com/3drepo/3drepo.io/assets/62481250/12665cdd-e706-404d-a3f5-c144a891f41f)

Reply:
And either way, one acts on empty strings and the other doesn't although they are similarly named?

Reply:
The description looks right to me, but I guess you are right, it could be renamed to make it clear that empty strings are treated as empty objects. Let's wait for Santiago to see if we still need the `removeEmptyObjcets` first, and then we deal with namings as well, ok?